### ISSUE TITLE
[J11] ASTParserFactory.provideCustomEnvironment(ASTParser) fails

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-xtend#804]
ASTParserFactory.provideCustomEnvironment(ASTParser) fails on Java 11

```
java.lang.ClassCastException: class jdk.internal.loader.ClassLoaders$AppClassLoader cannot be cast to class java.net.URLClassLoader (jdk.internal.loader.ClassLoaders$AppClassLoader and java.net.URLClassLoader are in module java.base of loader 'bootstrap')
	at org.eclipse.xtend.core.javaconverter.ASTParserFactory.provideCustomEnvironment(ASTParserFactory.java:181)
	at org.eclipse.xtend.core.javaconverter.ASTParserFactory.createJavaParser(ASTParserFactory.java:165)
	at org.eclipse.xtend.core.javaconverter.JavaConverter.toXtend(JavaConverter.java:108)
```

=> we need likely another means to provide a urlclassloader

**Comment by cdietrich** [comment]
cc @szarnekow

**Comment by cdietrich** [comment]
Fixed in 2.18
